phpunit.xml.dist

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
+         beStrictAboutChangesToGlobalState="true"
+         beStrictAboutOutputDuringTests="true"
+         beStrictAboutTodoAnnotatedTests="true"
+         bootstrap="tests/bootstrap.php"
+         colors="true"
+         verbose="true"
+>
+    <testsuites>
+        <testsuite name="Tests">
+            <directory>tests</directory>
+        </testsuite>
+    </testsuites>
+
+    <filter>
+        <whitelist>
+            <directory suffix=".php">./src</directory>
+        </whitelist>
+    </filter>
+
+    <php>
+        <env name="CLICKHOUSE_HOST" value="127.0.0.1:28123" />
+        <env name="CLICKHOUSE_USER" value="default" />
+        <env name="CLICKHOUSE_PASSWORD" value="" />
+        <env name="CLICKHOUSE_DATABASE" value="default" />
```

Comment:
tyhle params jsou meeh, nemohl bys to upravit ať to příjmá standardní dsn? jako postgres ... ať nemusíme pro clickhouse mít 5 env variables ...

Reply:
tomu jsem se chtel vyhnout. hlavne kvuli moznejm prazdnejm values a taky se mi moc nelibi zacinat custom dsn jako `http://`. vypada to pak jak endpoint, co se vola, i kdyz se to realne zparsuje a pak se to nevola. ale `clickhouse://` zas pouzit nemuzu, protoze protokol je dulezitej

Reply:
proč je důležitej v tomhle kontextu? s clickhouse:// mi to právě dává smysl.

a co myslíš těmi prázdnými values? to nechápu

Reply:
protoze prece chces definovat, jestli volas http nebo https

Reply:
To jde? A ok, jak to definuješ teď?

Reply:
proste volas klasicky http requesty, neni to nic jako postgres nebo redis  
username ani password nejsou povinny, proto je posilam jako query parameter, pokud jsou vyplneny. takze mi nedavalo smysl je zamichat do dsn abych je z toho zas rozparsovaval zpatky do jednotlivejch hodnot

Reply:
Zvenku ale nevoláš žádný requesty, používáš tvoje API, ne? Tak jak tomu API řekneš, aby to používalo HTTP místo HTTPS?

Reply:
tomu clientovi to musim rict pri konfiguraci, ne? http/https je soucasti endpointu

Reply:
no ale jako "endpoint" tu konfiguruješ jen host a port, tak proto se ptám kde je přesně to místo kde to nastavíš

Reply:
to tu konfiguruju blbe no... :D toho jsem si nevsim, to pak funguje jen diky nejaky psr implementaci, ze to defaultuje na http

Reply:
fixed

Reply:
tak to byl bordel v test traite

Reply:
tak můžeš pls napsat example jak to nakonfiguruješ? nebo odkaz na kód?

Reply:
koukam ze je to v readme jak instanciovat client

Reply:
takže nastavení připojení k serveru se vlastně zadává na 3 místech ... to je dost solidní bordel.

Reply:
vidim jen jedno misto a to je client. no?

Reply:
No, 3 místa při instancování CH clienta. Custom setup httpClienta, abys nastavil ten http/https (jestli to chápu dobře), pak jeden parametr pro host:port, a pak jako random prvky pole "default parameters" (což je imo největší prasárna).

Celé nastavení "připoj se k X:Y přes Z do databáze A pomocí usera B a hesla C" by mělo být jeden objekt/metoda, kde se tohle všechno předá a nějak už se to dostane do vnitřností tý tvojí libky. A hádej proč se na to používá dsn většinou - je to simple a obsahuje to tyhle všechny věci co potřebuješ ....

Reply:
nemel jsem v docs schema

do novyho instance clienta pres endpoint `http://x:y` + pripadne defaultni query params, kde muze bejt i auth

> Celé nastavení "připoj se k X:Y přes Z do databáze A pomocí usera B a hesla C" by mělo být jeden objekt/metoda

ja chci presne kopirovat jak to http interface funguje a ne si prave vymejslet vlastni abstrakci nad tim. kdyz budes chtit, klidne si v endpointu muzes mit i login `http://u:p@x:y`

ale cely je to zalozeny na otm, ze na jeden endpoint posilam requesty. a ty requesty muzou mit query params. takze tu na to mam proste dva argumenty a delej si co chces

chytrou abstrakci zkusil vymyslet ten smi2 client, coz dopadlo tak, ze jsi napr. v nekterejch pripadech nemohl pustit insert, protoze se to preplo do readonly mode :D a nikdo se v tom nevyzna, protoze je tam hromada side effects